lib/libzrepl/mgmt_conn.c

```diff
@@ -1168,8 +1168,8 @@ uzfs_zinfo_rebuild_from_clone(zvol_info_t *zinfo)
 	in_hdr.opcode = ZVOL_OPCODE_PREPARE_FOR_REBUILD;
 	in_hdr.io_seq = 0;
 
-	rc = uzfs_zvol_mgmt_get_handshake_info(&in_hdr, zinfo->name, zinfo,
-	    &out_hdr, &mack);
+	rc = uzfs_zvol_mgmt_get_handshake_info(&in_hdr,
```

Comment:
`name` passed here is to populate the name of volume on helping replica side. This is used to get the zinfo on the helping replica side.
Populating with clone name will not give zinfo on helping replica side, and, rebuild will fail.
In this particular name, same name will be populated to dw_volname also. This will be checked against the volname on degraded replica itself. So, degraded replica itself will NOT trigger rebuild.

Reply:
right. fixed it.